### ISSUE TITLE
Support for Bitpay/BIP70 payment protocol

### ISSUE THREAD

**Issue by imaginaryusername** [fyookball/electrum#590]
Was trying to copy paste payment protocol request from eGifter who was using Bitpay payment protocol like `bitcoincash:?r=https://bitpay.com/i/XXXXXXxxxxxxxXXXXXxxxxx` into the "send" field, nothing happened; since Electrum does support the BIP70 protocol, is this expected to be added on Electron-cash desktop/android as well?

Android 3.1.5 doesn't do anything, neither does desktop linux ver. 3.1.2 (Fedora package).

**Comment by imaginaryusername** [comment]
Never mind, seems like a duplicate of #581 .

**Comment by imaginaryusername** [comment]
Reopening as I tried on Bitpay again (3.1.5 post #581 ):

I tried paying a Bitpay invoice and got: 

`pay_script() takes 2 positional arguments but 3 was given` popup, the tx was broadcasted anyway - on the invoice page it returned 

`This invoice received a payment from a wallet that does not support payment protocol`

I'm not sure what was wrong, currently getting Bitpay to either confirm payment or refund me.

**Comment by imaginaryusername** [comment]
@kyuupichan I tested the latest githead against Bitpay invoice, can confirm it now works. Thanks for the hard work, can't wait for release (and hopefully Fedora repo).

**Comment by kyuupichan** [comment]
That's good, thanks for letting us know.

**Comment by protonn** [comment]
My payment was rejected by Bitpay. I built from latest HEAD

"This invoice received a payment from a wallet that does not support payment protocol." 

Transaction was broadcasted, waiting on refund.

electrum-3.1.5.7-g44498be.exe

**Comment by imaginaryusername** [comment]
Probably related to building for Windows. I ran straight from `./electron-cash` from the Linux command line (Fedora 27) and it was fine. @fyookball probably knows better?

**Comment by jhaand** [comment]
I had the same problem using 3.1.6 on Android today. The payment was suddenly expired once the transaction was sent.
I used the demo page from Bitpay to support the EFF. ( https://bitpay.com/demos )

**Comment by jhaand** [comment]
After a couple of hours, the invoices show as confirmed. 
I opened a ticket for one of the payments and see what will happen.
